app/src/main/java/com/x8bit/bitwarden/data/platform/manager/BiometricsEncryptionManagerImpl.kt

```diff
@@ -88,16 +92,21 @@ class BiometricsEncryptionManagerImpl(
     }
 
     /**
-     * Generates a [SecretKey] from which the [Cipher] will be generated.
+     * Generates a [SecretKey] from which the [Cipher] will be generated, or `null` if a key cannot
+     * be generated.
      */
-    private fun generateKey(): SecretKey {
+    private fun generateKeyOrNull(): SecretKey? {
         val keyGen = KeyGenerator.getInstance(
             KeyProperties.KEY_ALGORITHM_AES,
             ENCRYPTION_KEYSTORE_NAME,
         )
         keyGen.init(keyGenParameterSpec)
-        keyGen.generateKey()
-        return requireNotNull(getSecretKey())
+        try {
+            keyGen.generateKey()
+        } catch (e: ProviderException) {
```

Comment:
I'm thinking we also need to catch `InvalidAlgorithmParameterException` in here now right, rather than in `getOrCreateCipher`? I'm assuming that's coming from both `generateKey` or `getSecretKey` right?

Reply:
Indeed!